automates/text_reading/src/main/resources/org/clulab/aske_automates/grammars/context.yml

```diff
@@ -0,0 +1,106 @@
+vars: org/clulab/aske_automates/grammars/vars.yml
+
+# todo: try to simplify & combine rules!!
+rules:
+
+  - name: during_context
+    label: Context
+    priority: ${priority}
+    type: token
+    example: "Previous studies indicated that maize height and LAI would reduce when suffering water deficit during the early growing stage."
+    pattern: |
+      (?<context> [lemma = "during"] [chunk = /B-NP|I-NP/]+)
+
+  - name: when_where_context
+    label: Context
+    priority: ${priority}
+    type: token
+    example: "Under the same mulching fraction (fm = 0.5), αb decreased by 8.5% when fsw decreased from 1.0 to 0.5."
+    pattern: |
+      (?<context> ([chunk = /B-SBAR/ & !word = "that"]|[lemma = /(when|where)/]) [chunk = /I-NP|B-NP/]+ (${PP+NP})* [chunk = /I-VP|B-VP/]+ ([tag = /CC/]? [chunk = /B-ADVP|I-ADVP|B-ADJP|I-ADJP|B-PRT/]+)* [chunk = /I-NP|B-NP/ & !tag = /WDT/]* ((${PP+NP})+ ([tag = /TO/] [chunk = /B-NP|I-NP/]+)?)? ([chunk = /B-VP/] [chunk = /B-PP/]? [chunk = /B-NP|I-NP/]* [chunk = /B-ADJP|I-ADJP/]* [chunk = /B-PP/]? [chunk = /I-NP|B-NP/]*)* ([tag = /CC/] [chunk = /B-NP|I-NP/]+ [chunk = /B-VP/] ${PP+NP})? ([word = ","] [tag = /CC/] [chunk = /B-VP|I-VP/]+ [chunk = /B-NP|I-NP/]* (${PP+NP})?)? ([word = "and"] [chunk = /B-NP|I-NP/]+ [chunk = /B-VP|I-VP/]+ [chunk = /B-ADJP|I-ADJP/]+)?)
+      |
+      (?<context> [lemma = /(when|where)/] [tag = /^NN/] [lemma = /<|>|=|≥|≤/] [tag = /CD/] [tag = /NN/]? ([chunk = /B-VP/] [chunk = /B-NP|I-NP/]+)?)
+      |
+      (?<context> [lemma = /when/] [word = "not"]? [chunk = /B-ADJP/])
+      |
+      (?<context> [lemma = "when"] [tag = /VBG/] [chunk = /B-NP|I-NP/]+)
+
+  - name: if_condition
+    label: Context
+    priority: ${priority}
+    type: token
+    example: "C and C* are heat transfer coefficients that depend on the reference height selected for T and u, and, if this height is not low enough, on the stability."
+    pattern: |
+      (?<context> [lemma = "if"] [tag = /DT/]? [tag = /JJ|^NN/]+ ([tag = /IN/] [tag = /DT/]? [tag = /CD|JJ/]? [tag = /^NN/])? [tag = /^VB/] [tag = /RB/]? [tag = /JJ|^VB/]? [tag = /RB/]? ([tag = /IN/] [tag = /DT/]? [tag = /JJ|^NN/]+)* ([tag = /IN/] [tag = /CD/] [tag = /TO/] [tag = /CD/])?)
```

Comment:
Oh, that was the first rule that I wrote for the _if_ phrase, and soon I realized that using chunks is much more efficient. So I added other rules using chunks (2nd and 3rd rules). I think the second rule would cover most of the cases, so it won't hurt to delete the first rule. I didn't delete the rule just in case, but I can do that before merging this PR.